flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -157,14 +216,25 @@ public boolean reachedEnd() throws IOException {
 	 * @throws java.io.IOException
 	 */
 	@Override
-	public OUT nextRecord(OUT tuple) throws IOException {
+	public Row nextRecord(Row row) throws IOException {
 		try {
-			resultSet.next();
-			if (columnTypes == null) {
-				extractTypes(tuple);
+			hasNext = resultSet.next();
```

Comment:
If `resultSet.next()` was called at the end of `open()` we should call it again at the end of `nextRecord()`